flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/rules/logical/PythonCorrelateSplitRule.java

```diff
@@ -188,12 +203,20 @@ private FlinkLogicalCalc createTopCalc(
 
 	private ScalarFunctionSplitter createScalarFunctionSplitter(
 		int primitiveLeftFieldCount,
-		ArrayBuffer<RexCall> extractedRexCalls,
-		boolean isJavaTableFunction) {
+		ArrayBuffer<RexNode> extractedRexNodes,
+		RexNode tableFunctionNode) {
 		return new ScalarFunctionSplitter(
 			primitiveLeftFieldCount,
-			extractedRexCalls,
-			isJavaTableFunction ? x -> PythonUtil.isPythonCall(x, null) : PythonUtil::isNonPythonCall
+			extractedRexNodes,
+			node -> {
+				if (PythonUtil.isNonPythonCall(tableFunctionNode)) {
+					return PythonUtil.isPythonCall(node, null);
+				} else if (PythonUtil.containsNonPythonCall(node)) {
+					return PythonUtil.isNonPythonCall(node);
+				} else {
+					return node instanceof RexFieldAccess;
```

Comment:
Maybe add some document about split RexFieldAccess from PythonCorrelate for the `PythonCorrelateSplitRule`?